### PR TITLE
Upgrade kubeadm's version of docker support

### PR DESCRIPTION
**Pull request by yuansisi**
**What this PR does / why we need it**:
I suggest  that the docker version can be updated to 17.09
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:
```
kubeadm: update the maximum validated docker version to 18.06
```

### CODE REVIEW

**Comment by rosti** [comment]
We'll have to check in the issue tracker, but sticking to 17.03 was because of problems, that users were having with newer versions than this.

**Comment by fabriziopandini** [comment]
I'm not aware of what is the docker version supported/validated by sig-node and by kubernetes test infrastructure; I'm also a little bit concerned about changing this at this time of the cycle...
/cc @timothysc @neolit123 

BTW, if we go for this change, this deserves a release note and we should plan for changing documentation as well.

**Comment by neolit123** [comment]
AFAIK SIG node have not verified 17.09 and they are not going to.
this leaves us in a spot where 17.03 is a default safe bet for us, but it presents a distro problem because 17.03 is old on some distros.

my vote would be to still recommend 17.03, but deferring to @timothysc for the final call.

**Comment by neolit123** [comment]
> BTW, if we go for this change, this deserves a release note and we should plan for changing documentation as well.

:100:

**Comment by rosti** [comment]
Unless SIG-node give us the go signal, we probably shouldn't do anything. The missing warning can cause trouble for users and difficult to trace problems...

On the other hand, we can investigate what SIG-node folks are doing in this case:
- Do they simply use newer Docker versions without formerly verifying them?
- Do they use other CRIs (containerd, CRI-O, ...)?
- Do they stick to older Docker and Linux distro versions to be able to run it?

**Comment by rosti** [comment]
@yuansisi I hope, that we can reopen this, once SIG-node gives us the go call.

**Review by neolit123 — COMMENTED** [review]
@yuansisi 
please update the version to 18.06.

**Comment by neolit123** [comment]
@yuansisi thanks. 
please squash the commits into one.

**Comment by timothysc** [comment]
@chuckha @vincepri  @ncdc - Do we have automated results for

**Comment by neolit123** [comment]
@yuansisi 
please add a release note in the first message instead of `NONE`:
```
kubeadm: update the maximum validated docker version to 18.06
```

/lgtm
/hold

**Comment by yuansisi** [comment]
@neolit123 Thanks

**Review by timothysc — APPROVED** [review]
/approve

**Comment by timothysc** [comment]
@neolit123 back to you for unhold

File: cmd/kubeadm/app/util/system/docker_validator_test.go
Comment (dixudx) [inline cmd/kubeadm/app/util/system/docker_validator_test.go]: update to `// Requires [1.11, 18.06]`.

File: cmd/kubeadm/app/util/system/types_unix.go
Comment (dixudx) [inline cmd/kubeadm/app/util/system/types_unix.go]: update to `// Requires [1.11, 18.06]`.

File: cmd/kubeadm/app/util/system/types_windows.go
Comment (dixudx) [inline cmd/kubeadm/app/util/system/types_windows.go]: update to `// Requires [18.06] or later`.

**Review by dixudx — COMMENTED** [review]
@neolit123 Seems a consensu has been made. Where can I get this info?

**Comment by dixudx** [comment]
@yuansisi  Please squash your commits. Thanks.

**Comment by dims** [comment]
@ncdc @timothysc Where can i see the results from the `NodeConformance on centos with 18.06` in our test grid?

File: cmd/kubeadm/app/util/system/docker_validator_test.go
Comment (neolit123) [inline cmd/kubeadm/app/util/system/docker_validator_test.go]: @yuansisi 
sorry but we've missed adding some changes to the tests.

make this test have `warn: false` instead of `warn: true`:
```
		{
			info: types.Info{Driver: "driver_2", ServerVersion: "17.06.0-ce"},
			err:  false,
			warn: true,
		},
```

add this new test in there too:
```
		{
			info: types.Info{Driver: "driver_2", ServerVersion: "18.06.0-ce"},
			err:  false,
			warn: false,
		},
```

**Comment by ncdc** [comment]
> Where can i see the results from the NodeConformance on centos with 18.06 in our test grid?

@dims we've run this through our CI but it's not integrated with testgrid atm

**Comment by timothysc** [comment]
I'm also talking with @mataneja today about getting automation for 18.06 in place.  Several independent verifications are green.

**Comment by dims** [comment]
@ncdc @timothysc ack. i will try to add coverage for 18.06 too (see #9444)

File: cmd/kubeadm/app/util/system/docker_validator_test.go
Comment (timothysc) [reply cmd/kubeadm/app/util/system/docker_validator_test.go]: @yuansisi - ping.  We're getting low on time now.

**Comment by dims** [comment]
looks like things are working fine (example http://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/68742/pull-kubernetes-node-e2e/104697/artifacts/tmp-node-e2e-681b872d-coreos-beta-1883-1-0-v20180911/ )

**Comment by timothysc** [comment]
thanks @dims !

**Comment by neolit123** [comment]
/lgtm
we can add the missing tests later.

**Comment by gabrielfsousa** [comment]
docker 18.06 will be the recommend for kubernetes 1.12 ?

**Comment by neolit123** [comment]
@gabrielfsousa yes.

**Comment by gabrielfsousa** [comment]
cool !!!!

**Comment by gabrielfsousa** [comment]
the changelog still is the old one 
"The validated docker versions are the same as for v1.10: 1.11.2 to 1.13.1 and 17.03.x"

**Comment by dims** [comment]
same as etcd, same issue https://github.com/kubernetes/kubernetes/issues/69216

**Pull request merged**
